microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,514 @@
+                if _entry:
+                    cmd_res = node.execute(
+                        f'sed -i -e "s/rootdelay=300/rootdelay=300 log_buf_len=200M resume={sw_uuid}/g" /etc/default/grub.d/50-cloudimg-settings.cfg',
+                        shell=True,
+                        sudo=True
+                    )
+                    log.info(f"{cmd_res.exit_code}: Updated the 50-cloudimg-settings.cfg with resume={sw_uuid}")
+                else:
+                    _entry = node.tools[Cat].read_with_filter(
+                        "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                        '^GRUB_CMDLINE_LINUX_DEFAULT',
+                        sudo=True
+                    )
+                    if _entry:
+                        node.execute(
+                            f"sed -i '/^GRUB_CMDLINE_LINUX_DEFAULT=/ s/\"$/ rootdelay=300 log_buf_len=200M resume='{sw_uuid}'\"/'  /etc/default/grub.d/50-cloudimg-settings.cfg", 
+                            shell=True, 
+                            sudo=True
+                        )
+                    else:
+                        node.tools[Echo].write_to_file(
+                            f'GRUB_CMDLINE_LINUX_DEFAULT="console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300 log_buf_len=200M resume={sw_uuid}"',
+                            "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                            sudo=True,
+                            append=True
+                        )
+
+                    log.info(f"Added resume={sw_uuid} in 50-cloudimg-settings.cfg file")
+                
+                # This is the case about GRUB_HIDDEN_TIMEOUT
+                _entry = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_HIDDEN_TIMEOUT=',
+                    sudo=True
+                )
+                if _entry:
+                    cmd_res = node.execute(
+                        'sed -i -e "s/GRUB_HIDDEN_TIMEOUT=*.*/GRUB_HIDDEN_TIMEOUT=30/g" /etc/default/grub.d/50-cloudimg-settings.cfg',
+                        shell=True,
+                        sudo=True
+                    )
+                    log.info(f"{cmd_res.exit_code}: Updated GRUB_HIDDEN_TIMEOUT value with 30")
+                else:
+                    node.tools[Echo].write_to_file(
+                        "GRUB_HIDDEN_TIMEOUT=30",
+                        '/etc/default/grub.d/50-cloudimg-settings.cfg',
+                        sudo=True,
+                        append=True
+                    )
+                    log.info("Added GRUB_HIDDEN_TIMEOUT=30 in 50-cloudimg-settings.cfg file")
+                
+                # This is the case about GRUB_TIMEOUT
+                _entry = node.tools[Cat].read_with_filter(
+                    "/etc/default/grub.d/50-cloudimg-settings.cfg",
+                    '^GRUB_TIMEOUT=',
+                    sudo=True
+                )
+                if _entry:
```

Comment:
There are sed tool too. And this pattern is used many times. It's better to add one method in sed like `substitue_or_append` to simplify it.